app/Controllers/MediaController.php

```diff
@@ -188,6 +188,50 @@ public function download(Request $request, Response $response, string $userCode,
         return $this->streamMedia($request, $response, $this->storage, $media, 'attachment');
     }
 
+
+    /**
+     * @param  Request  $request
+     * @param  Response  $response
+     * @param  string  $vanity
+     * @param  string  $id
+     *
+     * @return Response
+     * @throws HttpNotFoundException
+     * @throws HttpBadRequestException
+     */
+    public function createVanity(Request $request, Response $response, int $id): Response
+    {
+        if (!$this->session->get('admin')) {
+            $media = $this->database->query('SELECT * FROM `uploads` WHERE `id` = ? LIMIT 1', $id)->fetch();
+        } else {
+            $media = $this->database->query(
+                'SELECT * FROM `uploads` WHERE `id` = ? AND `user_id` = ? LIMIT 1',
+                [$id, $this->session->get('user_id')]
+            )->fetch();
+        }
+
+        $data = $request->getParsedBody();
+        $vanity = $data['vanity'];
+        $vanity = strtolower(preg_replace('/[^a-z0-9]+/', '-', $vanity));
+
+        if (!$media) {
+            throw new HttpNotFoundException($request);
+        }
+
+        if ($vanity === '' || $media->code === $vanity) {
+            throw new HttpBadRequestException($request);
+        }
+
+        $this->database->query(
+            'UPDATE `uploads` SET `code` = ? WHERE `id` = ?',
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** here is missing handling of collisions, so what happen if you try to set the same vanity url to multiple uploads?‏

Reply:
I doubled checked, I already implemented a check for collisions, including if the user tried to create an empty vanity url. Here is where the check for this is:

https://github.com/SergiX44/XBackBone/pull/507/files#diff-1f64a59ac49e401b085fe29e9b0d34bb824ebd81e54c8416c2e0e2df674b14a3R221-R223

Reply:
Ah, that was my bad. I was doing a check on the same media object for collisions and not on all of then. Fixed :+1: